### PR TITLE
fix(auth): cache missing, valid Stripe plans on product.updated webhooks

### PR DESCRIPTION
**Pull request by biancadanforth**
Because:

* It's possible to have a valid Stripe plan missing from Redis on a product.updated webhook, for example, if the plan was created before the product metadata was valid and never updated afterward.

This commit:

* Loops through any valid plans for the product on a product.updated webhook and adds them to the Redis cache if they are missing.

Closes #FXA-5967

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts [hunk 1]
```diff
@@ -828,6 +833,18 @@ export class StripeWebhookHandler extends StripeHandler {
...
         }
       }
+      // Add any valid plans found in Stripe that are missing from the cache.
+      // This is possible e.g. if a plan was created before a product's metadata
+      // was valid and was never updated afterward.
+      for (const plan of latestStripePlansForProduct) {
+        const cachedPlan = updatedPlans.find((p) => p.id === plan.id);
+        if (!cachedPlan) {
+          updatedPlans.push({
+            ...plan,
```
Comment (StaberindeZA) [inline packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts:843]: Do we need to run the `subscriptionProductMetadataValidator` for the plan pulled from Stripe, similar to how the cachedPlans are validated with the product on lines 803-806?

```
await subscriptionProductMetadataValidator.validateAsync({
  ...product.metadata,
  ...plan.metadata,
});
```

**Review by StaberindeZA — CHANGES_REQUESTED** [review]
Just left one in line comment. Let me know what you think.

File: packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts (hunk 1 above)
Comment (biancadanforth) [reply packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts:843]: They're already validated in `fetchAllPlans` above.

Comment (StaberindeZA) [reply packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts:843]: Ah nice, I didn't realize `fetchAllPlans` validates the plan.

That said, I believe `subscriptionproductMetadataValidator` validates the Product metadata, with the Plan metadata, for each of the plans belonging to the Product.

At the moment validation is done for each of the `cachedPlans`, but not for any of the plans that are in Stripe, but missing from  `cachedPlans`.

Comment (biancadanforth) [reply packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts:843]: > At the moment validation is done for each of the `cachedPlans`, but not for any of the plans that are in Stripe, but missing from `cachedPlans`.

Validation is done for all of the plans in Stripe, including those not in the cache thanks to [`fetchAllPlans`](https://github.com/mozilla/fxa/blob/0ce0b6848055b776031ea20983c100a759907778/packages/fxa-shared/payments/stripe.ts#L322) (by way of [`validatePlan`](https://github.com/mozilla/fxa/blob/0ce0b6848055b776031ea20983c100a759907778/packages/fxa-auth-server/lib/payments/stripe.ts#L247) which it calls).

Comment (StaberindeZA) [reply packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts:843]: Ah, I see now. Thank you for walking me through it quickly.

**Review by StaberindeZA — APPROVED** [review]
LGTM! TY

File: packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts (hunk 1 above)
Comment (biancadanforth) [reply packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts:843]: Thank you for taking a careful look to make sure it made sense to you as well.

**Pull request merged**